### PR TITLE
Make invoke constexpr available in all modes

### LINKED ISSUE #4080: Should we make `invoke` `constexpr` in all modes?
MSVC STL (non-conformingly) makes `std::invoke` available in C++14 mode, and makes it `constexpr` since C++17 mode.

Does it make sense to leave `std::invoke` non-`constexpr` in C++14 mode? IIUC only non-conforming programs that use `std::invoke` in C++14 mode will be affected if we make `std::invoke` unconditionally `constexpr`.

### PR DESCRIPTION
**Pull request by grcm10**
Resolves #4080 by replacing the _CONSTEXPR17 with constexpr for std::invoke.

### CODE REVIEW

File: tests/std/tests/Dev11_0535636_functional_overhaul/test.cpp
Comment (StephanTLavavej) [inline tests/std/tests/Dev11_0535636_functional_overhaul/test.cpp:1054]: This comment should be updated.

Additionally, `<yvals_core.h>` has comments that need to be updated.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! I pushed comment changes and I believe this needs only one maintainer approval.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for improving `invoke`'s usability even in the ancient past of C++14! :calendar: :tada: :smile_cat:

**Pull request merged**
